ui_update: Use same @ symbol everywhere.

whilstsomebody opened the pull request:
This PR updates the @ icon across various locations in the app to ensure consistency.

Icon used :
```html
<i class="zulip-icon zulip-icon-at-sign" aria-hidden="true"></i>.
```

**Tasks completed:**

- [ ] Replace @ character in mention pills
- [x] Update @ in /#settings/notifications
- [x] Update "Copy mention syntax" icon in user popovers
- [x] Standardize unread mentions indicator icon in the left sidebar
- [x] Update @ indicator for the topics in conversation view. 

> **Note**
> - The @ before "Mention" in the sidebar did not require any changes. It already has the required updates.
> - The @ character for mention pills needs some extra work and can be fixed in a separate PR.

**Screenshots**


<table>
<tr>
<th>Before</th>
<th>After</th>
</tr>
<tr>
<td colspan="2"> /#settings/notifications </td>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/c7767d33-0b6f-4aaa-aba6-aac91a41ecea">
</td>
<td>
<img src="https://github.com/user-attachments/assets/928328b2-a2dc-4f53-a6f4-f9600755f649">
</td>
</tr>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/70502021-7598-453e-98b1-dea16001383c">
</td>
<td>
<img src="https://github.com/user-attachments/assets/b80799f8-d956-43c1-ad6d-233ef703a6b3">
</td>
</tr>
<tr>
<td colspan="2"> Copy Mention Syntax </td>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/a856dca1-8b5b-4a0a-a427-e4efbd627f0c">
</td>
<td>
<img src="https://github.com/user-attachments/assets/c1207b65-5d49-42f0-b116-f7e2cec7b4dd">
</td>
</tr>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/018bddd0-2ae5-4d90-a8e9-79d2cedfbe0d">
</td>
<td>
<img src="https://github.com/user-attachments/assets/50519cca-8a45-4358-a128-4fe33b4dd652">
</td>
</tr>
<tr >
<td colspan="2"> Left Sidebar @ Indicator</td>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/a29c83bc-f4a0-47db-8895-c3ea6ff7e3c3">
</td>
<td>
<img src="https://github.com/user-attachments/assets/18056874-d699-46f1-8c5f-14d01fa3bc87">
</td>
</tr>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/9ef7efb0-5feb-4000-a464-249cbfa588e5">
</td>
<td>
<img src="https://github.com/user-attachments/assets/51dd3c45-be23-466c-9f92-2900387f8373">
</td>
</tr>
<td colspan="2"> @ indicator for topics in inbox view</td>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/ae409e8e-cbf7-40c5-8048-e520e624ca97">
</td>
<td>
<img src="https://github.com/user-attachments/assets/6edc7314-c9e3-477d-a7ff-a07665f3f6af">
</td>
</tr>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/0c6b8123-cee3-4c39-a28d-40d3382c470a">
</td>
<td>
<img src="https://github.com/user-attachments/assets/04eb3192-400a-4376-b3b7-28dff61c4688">
</td>
</tr>
</table>


Partially Fixes : #22816 .

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

whilstsomebody commented:
Hey @alya , the PR is ready for review. Can you take a look?

whilstsomebody commented:
hey @alya , the PR is ready for a review. Can you take a look?

karlstolley commented on web/styles/app_components.css:
```diff
@@ -626,7 +626,13 @@ input.settings_text_input {
     margin-right: 5px;
     margin-left: 2px;
-    opacity: 0.5;
+    opacity: 0.7;
```
Why is the opacity being changed here?

whilstsomebody replied:
@karlstolley, I have not made any changes to the opacity. Before the latest update, which I made to resolve conflicts, the opacity was 0.7, as you can see [here](https://github.com/zulip/zulip/compare/2f3d3a8a81be5825c26401f2949f0c9b836ed32c..4ecee575c5888d68c331ab4d6304deb382862a44). The latest commits updated the opacity to 0.5, which, in my opinion, is too light for the @ icon. Therefore, I kept it at 0.7 instead of 0.5.

You can see the difference below:

At 0.7 opacity

![image](https://github.com/user-attachments/assets/8187707b-7576-483f-8d97-bb00ddd17f0d)


At 0.5 opacity

![image](https://github.com/user-attachments/assets/ad162f3f-8e68-4566-9c7f-9a60f19e3780)

karlstolley commented on web/styles/app_components.css:
```diff
@@ -626,7 +626,13 @@ input.settings_text_input {
+
+    .zulip-icon {
+        font-size: inherit;
+        position: relative;
```
I'd really like us to get away from using positioning and offsets to position things. Are there better ways outside of `app_components.css` to work with or within grid or flexbox?

whilstsomebody replied:
You’re right, @karlstolley. However, my reasoning is that many similar icons are positioned using specific positioning and offsets. I felt this approach would keep things consistent with the existing code.
Also, this time, I didn’t make any changes to app_components.css. Instead, I wrote local CSS for the icons directly in their respective files.
Thanks!

karlstolley commented on web/styles/settings.css:
```diff
@@ -633,6 +633,12 @@ input[type="checkbox"] {
             word-break: break-all;
         }
+
+        .zulip-icon {
```
As above, this is probably too generic a selector. I'm also not understanding where the `font-size` calculation is coming from. As above, you're using positioning here, but your offset is expressed in pixels, not ems.

whilstsomebody replied:
I have changed the font-size to 1em (16px) equal to font size of 'all" .

karlstolley replied:
I don't understand what that accomplishes, and the other feedback I've left has not been addressed.

whilstsomebody replied:
I've updated the font size of '@' before 'all' in /#settings/notifications. Since the font size of 'all' is 16px (1em), I set the font size of '@' to 1em as well.

My earlier calculation was based on observation. I also reviewed the font sizes of other icons, which initially influenced my approach.

karlstolley replied:
I don't think it's necessary to set `font-size: 1em`, as that's the size that will be inherited anyway.

whilstsomebody replied:
okh, removed the font-size.

whilstsomebody commented:
Hey @karlstolley , I have made some changes. Could you please take a look?
Please tell if I am mistaken somewhere.
Thanks!

whilstsomebody commented:
Hey @karlstolley , I have made some changes. Could you please take a look?
Please tell if I am mistaken somewhere.
Thanks!

whilstsomebody commented:
Hey @karlstolley , I have made some changes in the code and addressed your review comments. Could you please take a look?

whilstsomebody commented:
Hey @karlstolley , I have removed the font-size for the @ symbol before "all". Could you please take a look?

whilstsomebody commented:
Hey @karlstolley, could you please take a look ?

karlstolley commented:
I don't know what you expect me to review. I keep getting notifications from you to review this PR, and you still have not addressed my feedback by fixing your code. There is no way we're merging a bunch of positioning stuff here or overly generic CSS selectors.

whilstsomebody committed:
ui_update: Update the @ symbol in `/#settings/notifications`.

This commit updates the @ symbol before the 'all'
in the table head of notification table.

Fixes a part of #22816.

whilstsomebody committed:
ui_update: Update the @ symbol at 'copy mention syntax'.

This commit updated the @ symbol before the
'copy mention syntax'.

Fixes a part of #22816.

whilstsomebody committed:
ui_update: Update the @ symbol in sidebar components.

Following areas have been updated:

-unread mention indicator,
-unread mention indicator for more topics

Following area does not need changes:

- @ before 'Mention' in the sidebar

These changes are related to each other hence put
in a single commit.

Fixes a part of #22816.

whilstsomebody commented:
Hey @karlstolley , sorry for the delay. I have updated the code as per your reviews and suggestions. This time I haven't used any kind of positioning and offsets. 
Could you please take a look?

whilstsomebody commented:
Everything looks good except for the alignment of the @ icon in the topics sidebar. I am unable to figure out how to fix it. Could you suggest a way to resolve this?

![image](https://github.com/user-attachments/assets/80615d0c-21a6-42bc-a223-78071557f1a4)

whilstsomebody commented:
Hey @karlstolley, could you please review the PR again? This time, I have made sure to address and adhere to all of your comments. I have used divs instead of offsets and positioning to position the icon.

karlstolley commented on web/styles/inbox.css:
```diff
@@ -351,7 +356,7 @@
             }
 
             .unread_mention_info {
-                grid-area: unread_mention_info;
```
This grid area should be defined for the `.mention-and-unread-count-container`

whilstsomebody replied:
Done!

whilstsomebody commented:
@karlstolley , the PR is ready for the next review.

whilstsomebody commented:
Hey @karlstolley , could you please take a look?

karlstolley commented:
Have you updated your screenshots up at the top of the PR since your latest rounds of changes, @whilstsomebody? That'll make this much easier to finish reviewing.

whilstsomebody commented:
@karlstolley , updated the screenshots in the description. Please take a look.

whilstsomebody commented:
Hi @karlstolley, it has been a while since the PR has not been reviewed. Could you please review it?
Thanks!

whilstsomebody commented:
Hi @karlstolley , I have updated the screenshots in the PR description as per the latest changes. Could you please take a look?

karlstolley commented on web/templates/settings/notification_settings.hbs:
```diff
@@ -17,8 +17,15 @@
-                        <i class="fa fa-question-circle settings-info-icon tippy-zulip-tooltip" data-tippy-content="{{t 'Whether wildcard mentions like @all are treated as mentions for the purpose of notifications.' }}"></i>
+                    <th rowspan="2" width="14%">
+                        <div class="notification-table-header-container">
+                            <div class="at-sign-and-all-container">
```
It is necessary to have this class? You might be able to resolve this by not leaving `all` as a bare text node; wrapping it in a span or some other element will help it participate in the flexbox as a DOM element, which is generally easier to manage than a bare text node.

whilstsomebody replied:
Umm, you are right @karlstolley , I was worried about this while working because I was not able to align @ and all. Wrapping `all` in a `<span>` fixed it and we don't need `at-sign-and-all-container` class anymore. 
Thanks for the help!

whilstsomebody replied:
![image](https://github.com/user-attachments/assets/94c97a8c-f984-4025-98ec-d255542d2eba)

karlstolley commented:
@alya I've got one technical question still outstanding, but it shouldn't interfere with your doing a QA/design pass here.

alya commented:
Looks good to me!

While we're here, could you add a commit changing the Inbox and Recent conversations mentions tooltip to "You have unread mentions in this conversation."?

whilstsomebody commented:
> Looks good to me!
> 
> @whilstsomebody While we're here, could you add a commit changing the Inbox and Recent conversations mentions tooltip "You have mentions" -> "You have unread mentions in this conversation."?

I have updated this one @alya .

![image](https://github.com/user-attachments/assets/962e7329-c70d-465b-aeed-2ff76fc76034)

![image](https://github.com/user-attachments/assets/6e9a9e8e-d32c-4bc3-9992-76f28e496127)

Could you please take a look?
Thanks!

whilstsomebody committed:
tooltip: Update the Inbox and Recent conversations mentions tooltip.

This commit changes the mentions tooltip text for Inbox and
Recent conversations from "You have mentions" to
"You have unread mentions in this conversation."

Fixes a part of #22816.

alya commented:
Screenshot looks good to me (I haven't re-tested).

whilstsomebody commented:
Hey @alya , I am pining you to know if this one could be marks for integration review.
Thanks.

alya commented:
That's a question for @karlstolley , not me.

karlstolley commented on web/src/ui_util.ts:
```diff
@@ -142,15 +142,15 @@ export function update_unread_mention_info_in_dom(
     if (!stream_has_any_unread_mention_messages) {
-        $unread_mention_info_span.hide();
-        $unread_mention_info_span.text("");
+        $unread_mention_info_div.addClass("hide");
```
Why can't we just toggle the `.hide` class? I could see that `.flex` class being a real problem, especially if layout in this area were to take another approach that didn't involve flex.

whilstsomebody replied:
Okh I will think of a better approach.

whilstsomebody replied:
For fixing this, I am now appending and removing the @ icon. 
I created a template file `web/templates/widgets/unread_mention_indicator_icon_widget.hbs` for the @ icon. 
For changes, see [here](https://github.com/zulip/zulip/pull/32218/files#diff-83063d6da714df65708a383a2d849a397df014424074c3ed97eae857bd05ae00R147-R153).

karlstolley commented on web/styles/inbox.css:
```diff
@@ -338,7 +343,8 @@
 
             .unread_mention_info {
-                grid-area: unread_mention_info;
+                grid-area: mention-and-unread-count-container;
```
This seems to be made up; I find no grid definition anywhere with `mention-and-unread-count-container` declared. The `unread_mention_info` area *is* defined, and I do not at all understand why you are messing with that here.

whilstsomebody commented:
@karlstolley , could you please take a look?

karlstolley commented:
I would appreciate your directly addressing my feedback. I don't want to have to hunt around in your code to confirm that you've made the adjustments requested.

whilstsomebody commented:
> I would appreciate your directly addressing my feedback. I don't want to have to hunt around in your code to confirm that you've made the adjustments requested.

Hey @karlstolley , I have addressed your feedback. Could you please take a look at the replies of the above comments? I have also linked the respected area.

whilstsomebody commented:
Hey @karlstolley , could you please take a look?

karlstolley commented:
This is going to have to wait until after the 10.0 release. If you're not confident you'll be able to be around to see it through, you're welcome to put the `completion candidate` label on it.

karlstolley commented:
@whilstsomebody can you please make a note when closing a PR like this? Seems #38131 is preferred, but please be explicit
